ereuse_devicehub/resources/device/views.py

```diff
@@ -152,6 +156,57 @@ def query(self, args):
         return query.filter(*args['filter']).order_by(*args['sort'])
 
 
+class DeviceMergeView(View):
+
+    """View for merging two devices
+    Ex. ``device/<id>/merge/id=X``.
+    """
+    class FindArgs(MarshmallowSchema):
+        id = fields.Integer()
+
+    def get_merge_id(self) -> uuid.UUID:
+        args = self.QUERY_PARSER.parse(self.find_args, request, locations=('querystring',))
+        return args['id']
+
+    def post(self, id: uuid.UUID):
+        device = Device.query.filter_by(id=id).one()
+        with_device = Device.query.filter_by(id=self.get_merge_id()).one()
+        self.merge_devices(device, with_device)
+
+        db.session().final_flush()
+        ret = self.schema.jsonify(device)
+        ret.status_code = 201
+
+        db.session.commit()
+        return ret
+
+    def merge_devices(self, base_device, with_device):
+        """Merge the current device with `with_device` by
+        adding all `with_device` actions under the current device.
+
+        This operation is highly costly as it forces refreshing
+        many models in session.
+        """
+        snapshots = sorted(filterfalse(lambda x: not isinstance(x, actions.Snapshot), (base_device.actions + with_device.actions)))
+        workbench_snapshots = [ s for s in snapshots if s.software == (SnapshotSoftware.Workbench or SnapshotSoftware.WorkbenchAndroid)]
+        latest_snapshot_device = [ d for d in (base_device, with_device) if d.id == snapshots[-1].device.id][0]
+        latest_snapshotworkbench_device = [ d for d in (base_device, with_device) if d.id == workbench_snapshots[-1].device.id][0]
+
+        for action in (base_device.actions + with_device.actions):
+            latest_snapshot_device.actions.append(action)
+
```

Comment:
for the properties `type`, `hid`, `manufacturer`, `model` and `chassis` can properly be assigned in the following way:

```
base_device.type = latest_snapshot_device.type
base_device.hid = latest_snapshot_device.hid
base_device.manufacturer = latest_snapshot_device.manufacturer
base_device.model = latest_snapshot_device.model
base_device.chassis = latest_snapshot_device.chassis 
```